dev-support/docker/pkg-resolver/install-common-pkgs.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#######
+# Install SpotBugs 4.2.2
+#######
+mkdir -p /opt/spotbugs &&
+  curl -L -s -S https://github.com/spotbugs/spotbugs/releases/download/4.2.2/spotbugs-4.2.2.tgz \
+    -o /opt/spotbugs.tgz &&
+  tar xzf /opt/spotbugs.tgz --strip-components 1 -C /opt/spotbugs &&
+  chmod +x /opt/spotbugs/bin/*
+
+#######
+# Install Boost 1.72 (1.71 ships with Focal)
+#######
+# hadolint ignore=DL3003
+mkdir -p /opt/boost-library &&
+  curl -L https://sourceforge.net/projects/boost/files/boost/1.72.0/boost_1_72_0.tar.bz2/download >boost_1_72_0.tar.bz2 &&
+  mv boost_1_72_0.tar.bz2 /opt/boost-library &&
+  cd /opt/boost-library &&
+  tar --bzip2 -xf boost_1_72_0.tar.bz2 &&
+  cd /opt/boost-library/boost_1_72_0 &&
+  ./bootstrap.sh --prefix=/usr/ &&
+  ./b2 --without-python install &&
+  cd /root &&
+  rm -rf /opt/boost-library
+
+######
+# Install Google Protobuf 3.7.1 (3.6.1 ships with Focal)
+######
+# hadolint ignore=DL3003
+mkdir -p /opt/protobuf-src &&
+  curl -L -s -S \
+    https://github.com/protocolbuffers/protobuf/releases/download/v3.7.1/protobuf-java-3.7.1.tar.gz \
+    -o /opt/protobuf.tar.gz &&
+  tar xzf /opt/protobuf.tar.gz --strip-components 1 -C /opt/protobuf-src &&
+  cd /opt/protobuf-src &&
+  ./configure --prefix=/opt/protobuf &&
+  make "-j$(nproc)" &&
+  make install &&
+  cd /root &&
+  rm -rf /opt/protobuf-src
+
+######
+# Install pylint and python-dateutil
+######
+pip3 install pylint==2.6.0 python-dateutil==2.8.1
```

Comment:
Are we sure this is already installed once we call this?

Reply:
I didn't quite understand your question. The `already installed` is a little confusing. Are you asking if we're sure about `pylint` and `python-dateutil` getting installed upon running this command?

Reply:
Is pip3 installed by now?

Reply:
Yes, we first install the necessary base package installation from the default toolchain for all the platforms.